// Copyright 2020 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";

message LogicCalculatorOptions {
  extend CalculatorOptions {
    optional LogicCalculatorOptions ext = 338731246;
  }
  // The logical operation to apply.
  enum Operation {
    AND = 0;
    OR = 1;
    XOR = 2;
  }
  optional Operation op = 1;

  // Whether to negate the result.
  optional bool negate = 2;

  // Optional bool input values.
  repeated bool input_value = 3;
}
